### PR TITLE
feat(pagination): smart pagination

### PR DESCRIPTION
**Pull request by bassrock**
## Summary
* Ensure we smartly paginate our results. The first page will be smaller so the user sees content and the next page will be larger. We also should only load in the first 500 items in a users list for now.

## References 
* IN-850

## Implementation Details
* Chose 500 items for now, will align with @nzeltzer and leadership later on for our launch size.

* I also updated our large lists to have new data, but didn't change the data within. I want to find a tool to do that for us instead of sitting manually doing that.

## Test Steps
* Load your list, look at logs and confirm first page is 15 items, subsequent is 30.

## PR Checklist:
- [x] Added Unit / UI tests
- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

### CODE REVIEW

File: PocketKit/Sources/Sync/Operations/FetchArchive.swift [hunk 1]
```diff
@@ -64,19 +61,19 @@ class FetchArchive: SyncOperation {
...
+        var pagination = PaginationSpec(maxItems: SyncConstants.Archive.firstLoadMaxCount, pageSize: SyncConstants.Archive.initalPageSize)
 
         repeat {
             let result = try await fetchPage(pagination)
 
             if case .started = initialDownloadState.value,
                let totalCount = result.data?.user?.savedItems?.totalCount,
                pagination.cursor == nil {
-                initialDownloadState.send(.paginating(totalCount: totalCount))
+                initialDownloadState.send(.paginating(totalCount: totalCount > pagination.maxItems ? pagination.maxItems : totalCount))
```
Comment (cyndichin) [inline PocketKit/Sources/Sync/Operations/FetchArchive.swift:72]: style: what about using `min(totalCount, pagination.maxItems)` so we can understand better that we want the smaller count?

File: PocketKit/Sources/Sync/PocketSource.swift [hunk 2]
```diff
@@ -215,14 +215,13 @@ extension PocketSource {
...
             initialDownloadState: initialSavesDownloadState,
-            maxItems: maxItems,
             lastRefresh: lastRefresh
         )
 
         enqueue(operation: operation, task: .fetchSaves(maxItems: maxItems), completion: completion)
     }
 
-    public func refreshArchive(maxItems: Int = 400, completion: (() -> Void)? = nil) {
+    public func refreshArchive(maxItems: Int, completion: (() -> Void)? = nil) {
```
Comment (cyndichin) [inline PocketKit/Sources/Sync/PocketSource.swift:224]: do we still want `maxItems` here?

File: PocketKit/Sources/Sync/SyncConstants.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
+//
+//  File.swift
+//  
+//
+//  Created by Daniel Brooks on 3/9/23.
+//
```
Comment (cyndichin) [inline PocketKit/Sources/Sync/SyncConstants.swift:6]: update with other `Source Code Form` header?

File: PocketKit/Sources/Sync/SyncConstants.swift [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+import Foundation
+
+struct SyncConstants {
```
Comment (cyndichin) [inline PocketKit/Sources/Sync/SyncConstants.swift:10]: awesome! maybe we want to add Search constants here too in the future

File: PocketKit/Tests/SyncTests/Support/FetchSavesTests.swift [hunk 5]
```diff
@@ -199,6 +198,9 @@ class FetchSavesTests: XCTestCase {
 
     func test_refresh_whenItemCountExceedsMax_fetchesMaxNumberOfItems() async throws {
         var fetches = 0
+        let pages = Int(ceil(Double((SyncConstants.Saves.firstLoadMaxCount - SyncConstants.Saves.initalPageSize) / SyncConstants.Saves.pageSize))) + 2
```
Comment (cyndichin) [inline PocketKit/Tests/SyncTests/Support/FetchSavesTests.swift:201]: shall we add tests for Archive as well?

**Review by Gio2018 — APPROVED** [review]
Looks good @bassrock , the log reported what you say in the description!

**Review by cyndichin — APPROVED** [review]
logs look good, I see initial 15 and subsequent 30. a couple of comments, but non blocker!

**Pull request merged**